x-pack/plugins/security_solution/public/common/components/event_details/summary_view.tsx

```diff
@@ -0,0 +1,232 @@
+    return (
+      <Ip
+        contextId={`alert-details-value-formatted-field-value-${contextId}-${eventId}-${fieldName}-${value}`}
+        eventId={eventId}
+        fieldName={fieldName}
+        value={value}
+      />
+    );
+  }
+
+  if (fieldName === 'signal.status') {
+    return (
+      <EuiBadge>
+        <FormattedFieldValue
+          contextId={`alert-details-value-formatted-field-value-${contextId}-${eventId}-${fieldName}-${value}`}
+          eventId={eventId}
+          fieldName={fieldName}
+          fieldType={fieldType}
+          value={value}
+        />
+      </EuiBadge>
+    );
+  }
+
+  return (
+    <FormattedFieldValue
+      contextId={`alert-details-value-formatted-field-value-${contextId}-${eventId}-${fieldName}-${value}`}
+      eventId={eventId}
+      fieldName={fieldName}
+      fieldType={fieldType}
+      value={value}
+      linkValue={linkValue}
+    />
+  );
+};
+
+export const SummaryViewComponent: React.FC<{
+  browserFields: BrowserFields;
+  data: TimelineEventsDetailsItem[];
+  eventId: string;
+  timelineId: string;
+}> = ({ data, eventId, timelineId, browserFields }) => {
+  const summaryList = useMemo(() => {
+    const ruleIdField = data.find((d) => d.field === 'signal.rule.rule_id');
+    const ruleId = getOr(null, 'values.0', ruleIdField);
+    return data != null
+      ? fields.reduce<Summary>((acc, item) => {
+          const field = data.find((d) => d.field === item.id || d.field === item.linkField);
+          if (!field) {
+            return acc;
+          }
+
+          const fieldValue = getOr(null, 'values.0', field);
+          const category = field.category;
+          const fieldType = get(`${category}.fields.${field.field}.type`, browserFields) as string;
+          const description = getDescription({
+            contextId: timelineId,
+            eventId,
+            fieldName: item.id,
+            value: fieldValue,
+            fieldType: item.fieldType ?? fieldType,
+            linkValue: item.id === SIGNAL_RULE_NAME_FIELD_NAME ? ruleId : undefined,
+          });
+
+          return [
+            ...acc,
+            {
+              title: item.label ?? item.id,
+              description,
+            },
+          ];
+        }, [])
+      : [];
+  }, [browserFields, data, eventId, timelineId]);
+
+  const messageData = (data || []).find((item) => item.field === MESSAGE_FIELD_NAME);
+  const message = get('values.0', messageData);
+  const [readMoreButtonText, setReadMoreButtonText] = useState(i18n.READ_MORE);
+  const [isOverflow, setIsOverflow] = useState<boolean | null>(null);
+  const [isExpanded, setIsReadMoreClicked] = useState<boolean | null>(null);
+  const descriptionRef = useRef<HTMLDivElement>(null);
+  const toggleReadMore = useCallback(() => {
+    setIsReadMoreClicked((prevState) => !prevState);
+    setReadMoreButtonText((prevState) =>
+      prevState === i18n.READ_MORE ? i18n.READ_LESS : i18n.READ_MORE
+    );
+  }, []);
+
+  useEffect(() => {
```

Comment:
could you extract that into a separate hook, so we can reuse that in other places?

Reply:
OK, I've isolated it to a component.